packages/@aws-cdk-testing/cli-integ/lib/with-cdk-app.ts

```diff
@@ -163,6 +163,36 @@ export function withCDKMigrateFixture(language: string, block: (content: TestFix
   return withAws(withTimeout(DEFAULT_TEST_TIMEOUT_S, withCdkMigrateApp(language, block)));
 }
 
+/**
+ * Retry wrapper that executes a test callback up to maxAttempts times
+ *
+ * If any attempt succeeds, it returns immediately. If all attempts fail,
+ * it throws the last error encountered.
+ */
+
+export function withRetry<T extends TestContext>(
+  callback: (context: T) => Promise<void>,
+  maxAttempts: number = 3,
```

Comment:
Not significant, but I'm wondering if 2 is a better default.

If a test fails once, it could be flaky; if it fails twice, the likelihood of it being actually flaky reduces significantly IMO (unless we have a known problem, in which case it should fail more than twice anyway). The third attempt may not bring much value here.